pkg/migrator/migrator.go

```diff
@@ -91,6 +95,37 @@ func deleteOpsrc(opsrc *v1.OperatorSource, client client.Client) error {
 	if err != nil {
 		return err
 	}
+
+	// delete child resources
+	objMeta := metav1.ObjectMeta{
```

Comment:
this will fix the issues for clusters upgrading from 4.5.z -> 4.6.z, but not clusters that are updating from 4.6.0 / 4.6.1 -> 4.6.z, right?

Reply:
I think this fix on marketplace in combination with the changes on OLM should resolve the issue on 4.6.z update.

Reply:
I don't believe that's the case Evan, this migrator runs whenever the marketplace operator starts up in 4.6, so it would fix the problem for both update cases.

Reply:
Ignore my last comment, I didn't realize this was iterating over the list of operator sources.

IMO we should actually just look at the deployments and services in the marketplace namespace with the right annotations and forcefully delete them.

Reply:
@kevinrizza I don't know if that's any different than what's being done here. This solution looks fine to me unless I'm missing something.

Reply:
Ignore my last comment, I'm +1 with @ecordell and @kevinrizza it'll be safer to just  wipe the namespace clean so that we don't have to depend on any OLM changes.